zerver/views/message_fetch.py

```diff
@@ -274,6 +277,15 @@ def by_streams(self, query: Query, operand: str, maybe_negate: ConditionTransfor
                 type_id__in=public_streams_queryset).values_list('id', flat=True).order_by('id')
             cond = column("recipient_id").in_(recipient_ids)
             return query.where(maybe_negate(cond))
+        if operand == 'web-public':
+            # TODO: Find a better way to get realm
+            realm =  get_user_profile_by_email("hamlet@zulip.com").realm
```

Comment:
How do I get realm here via the request?

Reply:
Via `zerver/lib/subdomain.py` code.

Reply:
zerver/lib/subdomains.py